File: Problems/FixTheCode/FTC_1_sorting-algorithms/source_python/insertionSorter.py
from customSorter import CustomSorter
from arrays import swapTwoNumbers

class InsertionSorter(CustomSorter):

    def sort(self):
        print("Sorting with Insertion Sort.")

        arraySize = len(self.array)
        for i in range(1, arraySize-2):
            j = i
            while j > 0 and self.array[j - 1] < self.array[j]:
                swapTwoNumbers(self.array, j, j - 1)
                j -= 1

File: Problems/FixTheCode/FTC_1_sorting-algorithms/java/source/InsertionSorter.java
public class InsertionSorter implements Sorter{

    public InsertionSorter(){}

    public int[] Sort(int[] array){
        System.out.println("Sorting with Insertion Sort.");

        int[] sortedArray = array.clone();

        for (int i=1; i<sortedArray.length-2; i++)
        {
            int j = i;
            while(j>0 && sortedArray[j-1] < sortedArray[j]) {
                Arrays.SwapTwoNumbers(sortedArray, j, j-1);
                j--;
            }
        }
        return sortedArray;
    }

}